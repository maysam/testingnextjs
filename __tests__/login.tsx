import React from 'react'
// import { shallow } from 'enzyme'
import { render } from '@testing-library/react'
import Login from '../pages/login'

describe('SomeComponent', () => {
  // const wrapper = shallow(<Login />) // Rendering
  // expect(wrapper.find('Result').length).toBe(1) // Has Result component
  // expect(wrapper.find('Result').prop('type')).toBe('success') // The type of the Result component is success
  const { container } = render(<Login />)
  expect(container.firstChild).toMatchSnapshot()
})

test('two plus two is four', () => {
  expect(2 + 2).toBe(4)
})

// test('Title for the test case', () => {
//   const value = 1
//   const expectedValue = 2
//   const result = fn(value)
//   expect(result).toBe(expectedValue)
// })
