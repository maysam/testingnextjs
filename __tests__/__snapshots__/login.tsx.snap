// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
    >
      <label
        class=""
        for="LoginForm_method"
        title="Login with"
      >
        Login with
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-16"
    >
      <div
        class="ant-form-item-control has-success"
      >
        <span
          class="ant-form-item-children"
        >
          <div
            class="ant-radio-group ant-radio-group-outline"
            id="LoginForm_method"
          >
            <label
              class="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
            >
              <span
                class="ant-radio-button ant-radio-button-checked"
              >
                <input
                  checked=""
                  class="ant-radio-button-input"
                  name="method"
                  type="radio"
                  value="email"
                />
                <span
                  class="ant-radio-button-inner"
                />
              </span>
              <span>
                Email
              </span>
            </label>
            <label
              class="ant-radio-button-wrapper"
            >
              <span
                class="ant-radio-button"
              >
                <input
                  class="ant-radio-button-input"
                  name="method"
                  type="radio"
                  value="mobile"
                />
                <span
                  class="ant-radio-button-inner"
                />
              </span>
              <span>
                Mobile Number
              </span>
            </label>
            <label
              class="ant-radio-button-wrapper"
            >
              <span
                class="ant-radio-button"
              >
                <input
                  class="ant-radio-button-input"
                  name="method"
                  type="radio"
                  value="nid"
                />
                <span
                  class="ant-radio-button-inner"
                />
              </span>
              <span>
                National ID
              </span>
            </label>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
    >
      <label
        class="ant-form-item-required"
        for="LoginForm_email"
        title="Email Address"
      >
        Email Address
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-16"
    >
      <div
        class="ant-form-item-control"
      >
        <span
          class="ant-form-item-children"
        >
          <span
            class="ant-input-affix-wrapper"
          >
            <span
              class="ant-input-prefix"
            >
              <i
                aria-label="icon: mail"
                class="anticon anticon-mail"
                style="color: rgba(0, 0, 0, 0.25);"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="mail"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                  />
                </svg>
              </i>
            </span>
            <input
              autocomplete="email"
              class="ant-input"
              data-__field="[object Object]"
              data-__meta="[object Object]"
              id="LoginForm_email"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
    >
      <label
        class="ant-form-item-required"
        for="LoginForm_password"
        title="Password"
      >
        Password
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-16"
    >
      <div
        class="ant-form-item-control"
      >
        <span
          class="ant-form-item-children"
        >
          <span
            class="ant-input-password ant-input-affix-wrapper"
          >
            <span
              class="ant-input-prefix"
            >
              <i
                aria-label="icon: lock"
                class="anticon anticon-lock"
                style="color: rgba(0, 0, 0, 0.25);"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="lock"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                  />
                </svg>
              </i>
            </span>
            <input
              action="click"
              autocomplete="current-password"
              class="ant-input"
              data-__field="[object Object]"
              data-__meta="[object Object]"
              id="LoginForm_password"
              name="password"
              placeholder="password"
              type="password"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <i
                aria-label="icon: eye-invisible"
                class="anticon anticon-eye-invisible ant-input-password-icon"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="eye-invisible"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                  />
                  <path
                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                  />
                </svg>
              </i>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-control-wrapper ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
    >
      <div
        class="ant-form-item-control"
      >
        <span
          class="ant-form-item-children"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="submit"
          >
            <span>
              Login
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</form>
`;
